packages/astro/test/fixtures/content-layer/src/content/config.ts

```diff
@@ -18,4 +18,52 @@ const dogs = defineCollection({
 		temperament: z.array(z.string())
 	}),
 })
-export const collections = { blog, dogs };
+
+const cats = defineCollection({
+	type: "experimental_data",
+	loader: async function() {
```

Comment:
Why do we define `async` if the function doesn't return any promise?

Reply:
Just so the fixture tests that promises work

Reply:
Do you know if it is mandatory? I saw that the types accept returning data that don't return a promise

Reply:
We `await` it [here](https://github.com/withastro/astro/pull/11386/files#diff-fff398c3bcb68ee6e2e9161735922902d26b20550bf701ca1dc745ce7baaef79R151), so it's supported but not mandatory.